### DISCUSSION TITLE
Shouldn't "horizontal" and "vertical" be exchanged in perpendicular coordinates?

### DISCUSSION THREAD

**Discussion by sijow** [cetz-package/cetz#581] [Q&A]
The [documentation](https://cetz-package.github.io/docs/basics/coordinate-systems.html#perpendicular) for perpendicular coordinates says

> `horizontal`: The coordinate through which the horizontal line passes.
> `vertical`: The coordinate through which the vertical line passes.
>
> You can use the implicit syntax of `(horizontal, "-|", vertical)` or `(vertical, "|-", horizontal)`.

while the actual behavior of the keywords seems like the other way round. Look at the example in the documentation:

```
line("p1.south-east", (horizontal: (), vertical: "xline.end"))
line("p2.south-east", ((), "|-", "xline.end")) // Short form
```
 
In the first line it's actually the horizointal line that goes through `xline.end``.

The short form looks right: `-` indicates a horizontal line going through `xline.end`.

**Comment by sijow** [comment]
I guess the root of the problem is that the keyword names `horizontal` and `vertical` are ambiguous, they can be understood as horizontal/vertical line (as documented) or horizontal/vertical coordinate (as implemented). Tikz uses the keys `horizontal line through` and `vertical line through` which remove the ambiguity but are very verbose.

**Reply by johannes-wolf** [reply]
Maybe something like `take-x`/`take-y`/`take-z` is more explanatory?

**Reply by sijow** [reply]
Would be much clearer but is it always the case that x and y refer respectively to the horizontal and vertical coordinates in cetz?

**Reply by sijow** [reply]
I see now that taking the `x` or `y` value is exactly what this coordinate system does (even when those don't correspond to horizontal or vertical directions) so `take-x`/`take-y`/`take-z` look like great names.

**Comment by michaelfortunato** [comment]
I had knuesel's interpretation too
